### PR TITLE
CB-3769-user-role-change-event

### PR DESCRIPTION
**Pull request by yagudin10**

### CODE REVIEW

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/BaseWebSession.java [hunk 1]
```diff
@@ -109,6 +109,15 @@ public synchronized boolean updateSMSession(SMAuthInfo smAuthInfo) throws DBExce
         return userContext.refresh(smAuthInfo);
     }
 
+    public synchronized void refreshUserData() {
+        try {
+            userContext.refreshAccessibleProjects();
+        } catch (DBException e) {
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/BaseWebSession.java:116]: we also should refresh permissions

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebUserContext.java [hunk 2]
```diff
@@ -123,14 +123,18 @@ public synchronized boolean refresh(
...
-            this.accessibleProjectIds.clear();
-            this.accessibleProjectIds.addAll(
-                Arrays.stream(rmController.listAccessibleProjects()).map(RMProject::getId).collect(Collectors.toSet())
-            );
+            refreshAccessibleProjects();
         }
         return isSessionChanged;
     }
 
+    public void refreshAccessibleProjects() throws DBException {
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebUserContext.java]: synchronized

File: webapp/packages/core-localization/src/locales/ru.ts [hunk 3]
```diff
@@ -109,5 +109,6 @@ export default [
   ['app_root_server_node_changed_title', 'Приложение было перезапущено'],
   ['app_root_server_node_changed_message', 'Приложение было перезапущено. Пожалуйста, перезагрузите страницу.'],
   ['app_root_quota_exceeded', 'Квота превышена'],
+  ['app_root_event_permissions_changed_message', 'Ваши настройки доступа были изменены'],
```
Comment (Wroud) [inline webapp/packages/core-localization/src/locales/ru.ts]: `Ваши права доступа были изменены`

**Review by Wroud — APPROVED** [review]

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebUserContext.java [hunk 4]
```diff
@@ -143,6 +147,11 @@ public synchronized void refreshSMSession() throws DBException {
             smCredentials.getSmSessionId(),
             smCredentials.getPermissions()
         );
+        this.securityController = application.createSecurityController(this);
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebUserContext.java]: we should not recreate clients

File: server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSSubjectPermissionUpdatedEventHandler.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        activeUserSession.refreshUserData();
+        super.updateSessionData(activeUserSession, event);
+    }
+
+    @Override
+    protected boolean isAcceptableInSession(@NotNull BaseWebSession activeUserSession, @NotNull WSSubjectPermissionEvent event) {
+        if (!super.isAcceptableInSession(activeUserSession, event)) {
+            return false;
+        }
+        var user = activeUserSession.getUserContext().getUser();
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSSubjectPermissionUpdatedEventHandler.java:47]: use subject type from event

**Review by alexander-skoblikov — APPROVED** [review]

File: webapp/packages/core-root/src/SessionPermissionsResource.ts [hunk 6]
```diff
@@ -7,20 +7,47 @@
...
       sessionDataResource,
       () => {},
       () => {},
     );
+
+    sessionPermissionEventHandler.onEvent<ISessionPermissionEvent>(
+      ServerEventId.CbSubjectPermissionsUpdated,
+      () => {
+        this.dataSynchronizationService
+          .requestSynchronization('permissions', this.localizationService.translate('app_root_event_permissions_changed_message'))
```
Comment (Wroud) [inline webapp/packages/core-root/src/SessionPermissionsResource.ts]: probably you can't use core-localization here (circular dependency between packages)

**Review by Wroud — CHANGES_REQUESTED** [review]

File: webapp/packages/plugin-root/src/DataSynchronization/DataSynchronizationNotification.tsx [hunk 7]
```diff
@@ -61,7 +61,7 @@ export const DataSynchronizationNotification = observer<NotificationComponentPro
             {groups.map(messages => {
               const message = uniq(messages.map(m => m.message)).join('\n');
 
-              return <div key={messages[0].label}>{message}</div>;
+              return <div key={messages[0].label}>{translate(message)}</div>;
```
Comment (Wroud) [inline webapp/packages/plugin-root/src/DataSynchronization/DataSynchronizationNotification.tsx]: probably you need to translate every message in messages in map function

**Review by Wroud — CHANGES_REQUESTED** [review]

**Review by Wroud — APPROVED** [review]

**Pull request merged**